### PR TITLE
XSUP-40766/ Autofocus V2

### PR DESCRIPTION
**Pull request by RotemAmit**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-40766)

## Description
* Added a client class that will perform the request with the build in retry mechanism.
* Added debug logs.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py [hunk 1]
```diff
@@ -1000,19 +1036,33 @@ def search_indicator(indicator_type, indicator_value):
...
+                headers=headers,
+                params=params
+            )
+        demisto.debug(f'search_indicator: {result.status_code=} {result=}')
 
-        result_json = result.json()
+        try:
+            result_json = result.json()
+        except ValueError:
+            result_json = {}
```
Comment (MLainer1) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py:1060]: Maybe add a debug message?

**Review by MLainer1 — COMMENTED** [review]
😎

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py (hunk 1 above)
Comment (RotemAmit) [reply Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py:1060]: There is a debug message in the next line.

**Review by eyalpalo — APPROVED** [review]

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py [hunk 2]
```diff
@@ -274,6 +276,93 @@
...
 
 
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    def __init__(self, url: str, verify: bool = True):
+        super().__init__(base_url=url, verify=verify)
+
+    def get_url_enrichment(self, headers: dict, params: dict, ok_codes: tuple) -> Response:
```
Comment (eyalpalo) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py:286]: Basically, we don't have to have these methods. we can call the HTTP request (add the relevant parameters to the method).

**Review by eyalpalo — CHANGES_REQUESTED** [review]

**Review by eyalpalo — APPROVED** [review]

**Pull request merged**